### ISSUE TITLE
Unknown identifier ClassDefEx when criticalPaths feature is enabled

### ISSUE THREAD

**Issue by matsbryntse** [bryntum/support#2749]
Extract ZIP to basic example folder, crashes only when 
```
"features": {
            "criticalPaths": {
                "disabled": false
            },
        },

```
[Archive.zip](https://github.com/bryntum/support/files/6369323/Archive.zip)
[Forum post](https://www.bryntum.com/forum/viewtopic.php?f=52&t=17304&p=85873#p85873)

Hi,
Hope you guys are doing well,
I have a dataset which when i load in gannt produces strange error complaining about depedency , and this error goes away if i disable feature "criticalPaths"
i am attaching the test case code i would be delighted if you can guide me what i am doing wrong , because this error comes up times to time and only way to remove that error is find the offending depedency and remove it and which is not very practical , so i was wondering if i can know what is causing this

**Comment by MRC38** [comment]
Hi Mats

Are you able to advise when this issue will be resolved please?

Thanks

**Comment by arcady-zherdev** [comment]
The root cause is the provided dataset is dirty. There are few tasks starting before the project actually starts. The Gantt does its best by aligning automatically scheduled tasks but it can't do anything w/ manually scheduled tasks at the moment.

Manually scheduled task w/ `id==13871` called `SOC GO LIVE` starts `2021-04-26 08:00:00` (beofre the project start date `2021-06-07 08:00:00`) and also forces its `CONSTRUCTION/FIT OUT` and `Fosse Park` parent tasks to start on the same date.

It seems critical paths feature is not ready for such datasets. We'll try to improve it to handle such cases gracefully.

**Comment by arcady-zherdev** [comment]
Similar issues: https://github.com/bryntum/support/issues/2247, https://github.com/bryntum/support/issues/1247

**Comment by arcady-zherdev** [comment]
This was fixed in the Gantt v4.2.4 as part of https://github.com/bryntum/support/issues/3127
